sisyphus/cleaner.py

```diff
@@ -230,7 +230,7 @@ def remove_directories(dirs, message, move_postfix='.cleanup', mode='remove', fo
             for i in tmp:
                 logging.info(i)
     else:
-        with tempfile.NamedTemporaryFile() as tmp_file:
+        with tempfile.NamedTemporaryFile(mode="w") as tmp_file:
             for directory in dirs:
                 tmp_file.write(directory + "\x00")
```

Comment:
I am wondering if this is the correct solution. Here we write a list of directories to a file. The directories should be delimited with zero-bytes `"\x00"` to work with the option `--files0-from` (cf. https://man7.org/linux/man-pages/man1/du.1.html)

If I see it correctly, the default mode is binary write and you change it to text write. With your proposal I think the lines are delimited with the characters "\x00". So then the du command below should not find the corresponding folders?

I think rather should we change what is written to binary:
```suggestion
        with tempfile.NamedTemporaryFile() as tmp_file:
            for directory in dirs:
                tmp_file.write(directory.encode() + b"\x00")
```
(But please test both variants)

Reply:
so with my variant I got some result (which is hard to confirm whether correct but looked reasonable)
will test yours now

Reply:
it seems that `\x` will make the next two characters be interpreted as a binary value. So your solution works fine.

Reply:
i just tested your variant aswell, both produce the same results. Feel free to merge whatever you prefer.